Shared/Objects/PlaybackBitrate/PlaybackBitrate.swift

```diff
@@ -0,0 +1,82 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import Defaults
+import Foundation
+import JellyfinAPI
+
+enum PlaybackBitrate: Int, CaseIterable, Defaults.Serializable, Displayable {
+    case auto = 0
+    case max = 360_000_000
+    case mbps120 = 120_000_000
+    case mbps80 = 80_000_000
+    case mbps60 = 60_000_000
+    case mbps40 = 40_000_000
+    case mbps20 = 20_000_000
+    case mbps15 = 15_000_000
+    case mbps10 = 10_000_000
+    case mbps8 = 8_000_000
+    case mbps6 = 6_000_000
+    case mbps4 = 4_000_000
+    case mbps3 = 3_000_000
+    case kbps1500 = 1_500_000
+    case kbps720 = 720_000
+    case kbps420 = 420_000
+
+    var displayTitle: String {
+        switch self {
+        case .auto:
+            return L10n.bitrateAuto
+        case .max:
+            return L10n.bitrateMax
+        case .mbps120:
+            return L10n.bitrateMbps120
+        case .mbps80:
+            return L10n.bitrateMbps80
+        case .mbps60:
+            return L10n.bitrateMbps60
+        case .mbps40:
+            return L10n.bitrateMbps40
+        case .mbps20:
+            return L10n.bitrateMbps20
+        case .mbps15:
+            return L10n.bitrateMbps15
+        case .mbps10:
+            return L10n.bitrateMbps10
+        case .mbps8:
+            return L10n.bitrateMbps8
+        case .mbps6:
+            return L10n.bitrateMbps6
+        case .mbps4:
+            return L10n.bitrateMbps4
+        case .mbps3:
+            return L10n.bitrateMbps3
+        case .kbps1500:
+            return L10n.bitrateKbps1500
+        case .kbps720:
+            return L10n.bitrateKbps720
+        case .kbps420:
+            return L10n.bitrateKbps420
+        }
+    }
+
+    static func filterValues(
```

Comment:
What's the purpose of this?

Reply:
This Filter Values allows a built in function to filter the a picker to only include certain values. The purpose of this is related to future Video Player Overlay work. The reason for this would be to only show valid bitrates on the Player Overlay that do not conflict with App-Wide Setting.

For example, if the App-Wide setting had a cap of 80mbps, having an option for 120mbps or maximum would conflict with App-Wide setting. To prevent confusion, you could set your picker to pull from the value below to only include bitrates that were valid for the App-Wide setting:

`let filteredValues = PlaybackBitrate.filterValues(lessThanOrEqual: .mbps80)`

I have provided an example of this filter in the screenshots found here:

https://github.com/jellyfin/Swiftfin/pull/1147#issuecomment-2230003733

Reply:
As a note, that comment image used a revision to include all bitrates when Auto was selected. If we don't want it, I won't include it but this was the revised version of this:

`    static func filterValues(
        lessThan max: PlaybackBitrate? = nil,
        lessThanOrEqual maxOrEqual: PlaybackBitrate? = nil
    ) -> [PlaybackBitrate] {
        self.allCases.filter {
            $0 == .auto || (
                ((max == .auto || maxOrEqual == .auto) && $0.rawValue >= 0) ||
                    (max != nil && $0.rawValue < max!.rawValue) ||
                    (maxOrEqual != nil && $0.rawValue <= maxOrEqual!.rawValue)
            )
        }
    }`

Reply:
I'm gonna have to think about how this is actually an "app-wide maximum" vs just acting as a default. This would also probably be different for music when we get to it. Lastly, I would just do a `PlaybackBitrate.allCases.filter { $0 </<= maxBitrate }` instead.

Reply:
If we want to allow the player overlay version of this (TBD) to override the App-Wide Setting this filter is not necessary. Alternatively, if we would rather just filter this ad-hoc, this filter is also not necessary.

This filter is only valuable if we want to filter bitrates so only valid ones appear on a picker AND we don't need to filter it ad-hoc per location it's used.

Completely your call on this! I just felt it would take me 5 minutes to add this now to save the trouble later.

Reply:
> Lastly, I would just do a `PlaybackBitrate.allCases.filter { $0 </<= maxBitrate }` instead.

Gotcha! So regardless of if this is the Default Bitrate or the Maximum Bitrate, we do not need this filter?

Reply:
This has been removed. For the time being, the maximum vs default bitrate conversion is not required since this is the only location that bitrate can be managed. When the player overlay bitrate is set, this can be revisited based on it's implementation there.

That is, if we want it to be maximum, we can filter the picker and if we want it to be default we just set the default. That decision can be made when 2 bitrate locations exist.

I will mark this as resolved as the code in question is no longer in this PR.

Reply:
A default bitrate would be just that: the default bitrate selected when we create the playback URLs. Users can still select an alternate bitrate, even a higher one, if desired.